pkgs/applications/misc/pattypan/default.nix

```
@@ -0,0 +1,59 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, unzip
+, jre
+, jdk
+, ant
+, makeWrapper
+, makeDesktopItem
+, copyDesktopItems
+, dconf
+}:
+
+let
+  desktopItem = makeDesktopItem {
+    desktopName = "Pattypan";
+    genericName = "An uploader for Wikimedia Commons";
+    categories = [ "Utility" ];
+    exec = "pattypan";
+    name = "pattypan";
+  };
+in
+stdenv.mkDerivation rec {
+  pname = "pattypan";
+  version = "22.03";
+  src = fetchFromGitHub {
+    owner = "yarl";
+    repo = "pattypan";
+    rev = "v${version}";
+    sha256 = "0qmvlcqhqw5k500v2xdakk340ymgv5amhbfqxib5s4db1w32pi60";
+  };
+
+  nativeBuildInputs = [ copyDesktopItems jdk ant makeWrapper ];
+  buildInputs = [ (lib.getLib dconf) ];
+
+  buildPhase = ''
+  export JAVA_TOOL_OPTIONS="-Dfile.encoding=UTF8"
+  ant
```

Indenttion is incorrect

```suggestion
    export JAVA_TOOL_OPTIONS="-Dfile.encoding=UTF8"
    ant
```

Also you're not taking care of the `pre`/`post` hooks.

---

> Also you're not taking care of the `pre`/`post` hooks.

Could you elaborate? I added them because desktop items only work with them. Is there another way to do this?

---

When you overwrite `buildPhase`, you should make sure that you still call `runHook preBuild` before the build phase, and `runHook postBuild` after the build.

This allows user to override the derivation to more easily customize the build if they want to.

---

Thanks for the explanation, I just realized that.